plugins/modules/docker_image.py

```diff
@@ -160,6 +160,13 @@
       - Note that image IDs (hashes) are not supported.
     type: str
     required: yes
+  pull_platform:
```

Comment:
We have `platform` as a suboption in `build`. I know at least now there are no suboptions for `pull` but maybe for consistency, this should be a `platform` as suboption in `pull` not as `pull_platform` option?

Reply:
Hmm, that would definitely make it easier to extend in the future. Let me try this out.

Reply:
I tried that in 2737cf4f15dfad5423e407239058a61d58035d0c. I also used `apply_defaults=True` for `build`, that should simplify the code. Let's see what CI says :)

Reply:
Ok, the failing tests reminded me why `apply_defaults=True` is not a good idea: it also enforces required fields, which is something we do not want here :)